### DISCUSSION TITLE
Question about profiles when building vs. starting a native image

### DISCUSSION THREAD

**Discussion by kjq** [quarkusio/quarkus#40144] [Q&A] [answered]
We have three profiles - `prod`, `kubernetes`, and `cluster` (parent of `kubernetes`)

When we perform the native build we use a Maven profile `-Pproduction` and that sets `<quarkus.profile>prod,kubernetes</quarkus.profile>`

So our Maven job looks like `mvn install -am -pl client-service -Pproduction,native`

The build and running the native service are fine but we see these log warnings on startup...

```
The profile 'cluster' used to build the native image is different from the runtime profile 'prod'
The profile 'prod' used to build the native image is different from the runtime profile 'cluster'
...
Profiles kubernetes,prod,cluster activated.
```

Is this a real issue with something we are doing or just the way the startup logs the multiple profiles?  I did not think I needed to explicitly pass in the profiles (i tested with something locally and it seemed to startup correctly).

[a comment by an automated account is not shown]

**Comment by geoand** [comment]
cc @radcortez

**Comment by radcortez** [comment, marked as answer]
The parent profile configuration is recorded with a different priority compared to the JVM mode. In cases of overlapping properties within the active profiles, it may provide different configurations. 

https://github.com/quarkusio/quarkus/pull/40500 will fix this.

**Reply by kjq** [reply]
Thanks!
